### PR TITLE
[RFC] [DeviceType] Add support for MAIA

### PR DESCRIPTION
**Pull request by kyule7**
This PR extends `DLDeviceType` to support Microsoft [MAIA devices](https://news.microsoft.com/source/features/ai/in-house-chips-silicon-to-service-to-meet-ai-demand/).

### CODE REVIEW

**Comment by tqchen** [comment]
Thanks @kyule7 . Would like to get a bit more background on about this device type, is there any public available information that describes the proposed device?

**Comment by kyule7** [comment]
> Thanks @kyule7 . Would like to get a bit more background on about this device type, is there any public available information that describes the proposed device?

I do not think there is any good public information about the device, so it is difficult to describe any background and details about it at the moment. Would extending DLDeviceType require any public knowledge about the device?

**Comment by tqchen** [comment]
@kyule7 usually yes, since one goal of DLPack to be stable and widely used, so we would like to maintain a list of public available devices for that purpose. DLExtDevice would serve as a good middle ground for plug in less mature device requirements

**Comment by kyule7** [comment]
> @kyule7 usually yes, since one goal of DLPack to be stable and widely used, so we would like to maintain a list of public available devices for that purpose. DLExtDevice would serve as a good middle ground for plug in less mature device requirements

I used DLExtDevice for experiment with a new device for functionality, and planned to extend dlpack first in order to extend the dependent frameworks for the device support. I will close this PR since I do not think any public information about the device is available currently.

**Comment by tqchen** [comment]
thanks @kyule7 you are more than welcome to bring up followups once it is publicly available

**Comment by kyule7** [comment]
Reopening it since Maia 100 is publicly announced.

File: include/dlpack/dlpack.h [hunk 1]
```diff
@@ -116,6 +116,8 @@ typedef enum {
   kDLWebGPU = 15,
   /*! \brief Qualcomm Hexagon DSP */
   kDLHexagon = 16,
+  /*! \brief Microsoft NPU */
+  kDLMsNpu = 17,
```
Comment (tqchen) [inline include/dlpack/dlpack.h:120]: should it be kDLMSNPU? just to followup other convention for abbrev

**Comment by tqchen** [comment]
gentle ping @kyule7 :)

File: include/dlpack/dlpack.h (hunk 1 above)
Comment (kyule7) [reply include/dlpack/dlpack.h]: Renamed it to kDLMAIA since the MAIA is the released name.

**Review by tqchen — APPROVED** [review]

**Comment by kyule7** [comment]
Could anyone help merging this PR?

**Review by tqchen — APPROVED** [review]

**Comment by tqchen** [comment]
Thanks @kyule7 this is now merged

**Comment by edgarriba** [comment]
@tqchen worth to release ?

**Comment by leofang** [comment]
There're a few changes from the Array API side that we're wrapping up, would be nice to hold off a bit 🙂

**Comment by kyule7** [comment]
> @tqchen worth to release ?

@edgarriba Would like to apply this change in the dependent framework repos (pytorch and onnxruntime), would there be any chance to have a new release including this PR change?

**Comment by leofang** [comment]
Hi, v1.0rc is out a few days ago, could you give it a try?
https://github.com/dmlc/dlpack/releases/tag/v1.0rc

**Pull request merged**